hw/top_earlgrey/sw/autogen/chip/top_earlgrey.rs

```diff
@@ -754,8 +756,32 @@ pub enum TopEarlgreyPlicPeripheral {
     Edn0 = 29,
     /// edn1
     Edn1 = 30,
-    /// \internal Number of PLIC peripheral
-    End = 31,
+}
+
+impl TopEarlgreyPlicPeripheral {
+    /// Number of PLIC peripheral
+    const NUMBER: usize = 31;
+    /// Enum first valid value
+    const FIRST: u32 = Self::Unknown as u32;
+    /// Enum last valid value
+    const LAST: u32 = Self::Edn1 as u32;
+}
+
+impl TryFrom<u32> for TopEarlgreyPlicPeripheral {
+    type Error = u32;
+    fn try_from(val: u32) -> Result<Self, Self::Error> {
+        type Enum = TopEarlgreyPlicPeripheral;
+        match val {
+            Enum::FIRST..=Enum::LAST => {
+                Ok(
+                    // SAFETY: Following code is correct because generated
+                    // enum have subsequent values assigned to variants.
+                    unsafe { core::mem::transmute(val) },
```

Comment:
I agree that this is correct.  I'm concerned that if we try to import this into Tock, they will point out that a full match expression with all variants will optimize to the same code without requiring an `unsafe` block.

@jrvanwhy, WDYT?

Reply:
If it will be necessary i can "explod" it to full list of all variants or wrap it all by macro. 
(in the worst case, both methods simultaneously)

I'd prefer to avoid the macro as it might create a problem later. 
Similarly, I would not like to leave the choice to the compiler.
Rust is a young language and the compiler may not always generate optimal code.

Reply:
The answer at https://stackoverflow.com/questions/71167454/rust-implement-try-from-for-u8-enum indicates the compiler would optimize it fine. I'd rather it be safe code.

I think using the [`num_enum`](https://crates.io/crates/num_enum) would be a better solution, though.

Reply:
For short enum and -O2 it make non difference.
https://rust.godbolt.org/z/rsjhE8M8r

Lets deliver "exploded" version. It don't drag any dependency and it is easy to read.